plugins/inputs/mysql/mysql_test.go

```diff
@@ -178,6 +183,146 @@ func TestMysqlDNSAddTimeout(t *testing.T) {
+		{"", fields{{"ssl_fips_mode", "2", "2", false}}, nil}, // TODO: map this to STRICT or vice versa using integers
+		{"", fields{{"ssl_fips_mode", "OFF", "OFF", false}}, nil},
+		{"", fields{{"ssl_fips_mode", "ON", "ON", false}}, nil},
+		{"", fields{{"ssl_fips_mode", "STRICT", "STRICT", false}}, nil},
+		{"", fields{{"use_secondary_engine", "OFF", "OFF", false}}, nil},
+		{"", fields{{"use_secondary_engine", "ON", "ON", false}}, nil},
+		{"", fields{{"use_secondary_engine", "FORCED", "FORCED", false}}, nil},
+		{"", fields{{"transaction_write_set_extraction", "OFF", "OFF", false}}, nil},
+		{"", fields{{"transaction_write_set_extraction", "MURMUR32", "MURMUR32", false}}, nil},
+		{"", fields{{"transaction_write_set_extraction", "XXHASH64", "XXHASH64", false}}, nil},
+		{"", fields{{"slave_skip_errors", "OFF", "OFF", false}}, nil},
+		{"", fields{{"slave_skip_errors", "0", "0", false}}, nil},
+		{"", fields{{"slave_skip_errors", "1007,1008,1050", "1007,1008,1050", false}}, nil},
+		{"", fields{{"slave_skip_errors", "all", "all", false}}, nil},
+		{"", fields{{"slave_skip_errors", "ddl_exist_errors", "ddl_exist_errors", false}}, nil},
+		{"", fields{{"gtid_mode", "OFF", int64(0), false}}, nil},
+		{"", fields{{"gtid_mode", "OFF_PERMISSIVE", int64(0), false}}, nil},
+		{"", fields{{"gtid_mode", "ON", int64(1), false}}, nil},
+		{"", fields{{"gtid_mode", "ON_PERMISSIVE", int64(1), false}}, nil},
+	}
+
+	for i, testCase := range testCases {
+		if testCase.name == "" {
+			testCase.name = fmt.Sprintf("#%d", i)
+		}
+
+		rows := sqlmock.NewRows(columns)
+		for _, field := range testCase.fields {
+			rows.AddRow(field.key, field.rawValue)
+		}
+
+		mock.ExpectQuery(globalVariablesQuery).WillReturnRows(rows).RowsWillBeClosed()
+
+		acc := &testutil.Accumulator{}
+
+		err = m.gatherGlobalVariables(db, "test", acc)
+		if !assert.NoErrorf(t, err, "err on gatherGlobalVariables (test case %q)", testCase.name) {
+			continue
+		}
+
+		for _, metric := range acc.Metrics {
+			assert.Equalf(t, measurement, metric.Measurement, "wrong measurement (test case %q)", testCase.name)
+
+			if testCase.tags != nil {
+				assert.Equalf(t, testCase.tags, tags(metric.Tags), "wrong tags (test case %q)", testCase.name)
+			}
+
+			for key, value := range metric.Fields {
+				foundField := false
+
+				for _, field := range testCase.fields {
+					if field.key == key {
+						assert.Falsef(t, field.observed, "field %s observed multiple times (test case %q)", key, testCase.name)
+						assert.Equalf(t, field.parsedValue, value, "wrong value for field %s (test case %q)", key, testCase.name)
+						field.observed = true
+						foundField = true
+						break
+					}
+				}
+
+				if !assert.Truef(t, foundField, "unexpected field %s=%v (test case %q)", key, value, testCase.name) {
+					continue
+				}
+			}
+		}
+
+		for _, field := range testCase.fields {
+			assert.Truef(t, field.observed, "missing field %s=%v (test case %q)", field.key, field.parsedValue, testCase.name)
```

Comment:
@fxedel,

Thanks for those updates and for taking the time to put this together! We had one final question regarding `assert` vs `require`. In all our other tests we try to prefer the use of `require` so we fail right away so it is very clear what is failing in tests.

I'm ready to +1 once this change is made.

Thanks!